common/scala/src/main/scala/whisk/core/entity/Exec.scala

```diff
@@ -302,21 +309,29 @@ protected[core] object Exec extends ArgNormalizer[Exec] with DefaultJsonProtocol
 
           manifest.attached
             .map { a =>
-              val jar: Attachment[String] = {
-                // java actions once stored the attachment in "jar" instead of "code"
-                obj.fields.get("code").orElse(obj.fields.get("jar"))
+              val code = obj.fields.get("code")
+              val binary: Boolean = code match {
+                case Some(JsString(c)) => isBinaryCode(c)
+                case _ =>
+                  obj.fields.get("binary") match {
+                    case Some(JsBoolean(b)) => b
+                    case _                  => false
+                  }
+              }
+              val attachment: Attachment[String] = {
+                code
```

Comment:
If I read this correctly, an undefined `code` results in a DeserializationException? If that's the case, why even bother checking for the "binary" field above if code is `None`? The binary field also feels redundant, is it ever used?

As I read it, this could be simplified to:

```scala
val (binary, attachment) = code match {
  case Some(JsString(c)) => 
    (isBinaryCode(c), attFmt[String].read(c))
  case _ => 
    throw new DeserializationException(s"'code' must be a string defined in 'exec' for '$kind' actions")
}
```

Am I missing something?

Reply:
Chatted with @chetanmeh, I was indeed missing something:

`code` can be a JsObject, that breaks my simplification of course.